lib/tictactoe.rb

```diff
@@ -1,58 +1,82 @@
 # frozen_string_literal: true
 
 class TicTacToe
-  def initialize(io:)
-    @board = Array.new(9, ' ')
+  def initialize(io:, board:)
+    @board = board
     @current_token = 'X'
     @io = io
   end
 
   def render_board
-    board_output = <<~BOARD_RENDER 
-    #{@board[0]}|#{@board[1]}|#{@board[2]}
+    board_output = <<~BOARD_RENDER
+    #{@board.get_info_at_position(position: 0)}|#{@board.get_info_at_position(position: 1)}|#{@board.get_info_at_position(position: 2)}
```

Comment:
These lines got pretty overwhelming to read, and seem to have some repetition both within each line and between lines. Honestly putting stuff down in a declarative template like you're doing here has some real strengths - it's simple from a code perspective, which is a big win. But there's potential to extract some logic here into helper methods perhaps, if you wanted to try that approach.